src/pocketmine/Server.php

```diff
@@ -1801,6 +1801,17 @@ public function broadcastPacketsCallback($data, array $identifiers){
 		}
 	}
 
+	public function timingsPasteCallback($name, $matches, $failed){
+		$sender = $name === "CONSOLE" ? $this->consoleSender : $this->getPlayerExact($name);
+		if($sender !== null){
+			if($failed){
+				$sender->sendMessage(new TranslationContainer("pocketmine.command.timings.pasteError"));
+				return;
+			}
+			$sender->sendMessage(new TranslationContainer("pocketmine.command.timings.timingsUpload", ["http://paste.ubuntu.com/" . $matches . "/"]));
+			$sender->sendMessage(new TranslationContainer("pocketmine.command.timings.timingsRead", ["http://" . $this->getProperty("timings.host", "timings.pmmp.io") . "/?url=" . $matches]));
+		}
+	}
```

Comment:
maybe actually in the command? Because a timing paste is only started from there. The current timings classes are not suitable for this method for sure.